pkg/runtime.v2/runtime.go

```diff
@@ -17,45 +17,55 @@ limitations under the License.
 package runtimev2
 
 import (
-	"errors"
 	"maps"
 
 	corev1 "k8s.io/api/core/v1"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 	kueuelr "sigs.k8s.io/kueue/pkg/util/limitrange"
 
 	kubeflowv2 "github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v2alpha1"
 )
 
-var (
-	errorDifferentGVK  = errors.New("the GroupVersionKinds are different between old and new objects")
-	errorObjectsAreNil = errors.New("old or new objects are nil")
-)
-
 type Info struct {
-	Obj         client.Object
+	// Labels and Annotations to add to the RuntimeJobTemplate.
 	Labels      map[string]string
-	PodLabels   map[string]string
 	Annotations map[string]string
-	Policy
-	TotalRequests map[string]TotalResourceRequest
+	// Original policy values from the runtime.
+	RuntimePolicy RuntimePolicy
+	// Trainer parameters to add to the RuntimeJobTemplate.
+	Trainer
+	// Scheduler parameters to add to the RuntimeJobTemplate.
+	*Scheduler
 }
```

Comment:
I keep the original runtime policies values in the Info object for now, since we can't pass the specific policies to the EnforceMLPolicy and EnforcePodGroupPolicy() APIs. Similar to what we do today: https://github.com/kubeflow/training-operator/blob/62a058db0c445db1b7e16572d83fdace56f59476/pkg/runtime.v2/runtime.go#L40

In the future, we will discuss how we can optimize the Info object structure.

cc @tenzen-y

Reply:
Yeah, that is at least a requirement. Let's revisit here later. I can take that in the next week.